src/hooks.js

```diff
@@ -0,0 +1,41 @@
+import { useState, useEffect } from 'preact/hooks'
+
+const fetchBody = async (url, fetchOpts) => {
+  const state = {}
+  try {
+    state.response = await fetch(url, fetchOpts)
```

Comment:
The first call return the headers with the status of the request only so, at this point we don't have the content, but some usefull info may reside in this part of the request